omnigibson/transition_rules.py

```diff
@@ -103,115 +297,332 @@ def __call__(self, obj):
-            "At least one of individual_filters or group_filters must be specified!"
+        Args:
+            filter_name (str): Name of the filter whose object candidates will be pruned based on whether or not
+                the state @state's value is equal to @val
+            state (BaseObjectState): Object state whose value should be queried as a rule condition
+            val (any): The value @state should be in order for this condition to be satisfied
+            op (function): Binary operator to apply between @state's getter and @val. Default is operator.eq,
+                which does state.get_value() == val.
+                Expected signature:
+                    def op(state_getter, val) --> bool
+        """
+        self._filter_name = filter_name
+        self._state = state
+        self._val = val
+        self._op = op
 
-        # Store the filters
-        self.individual_filters = dict() if individual_filters is None else individual_filters
-        self.group_filters = dict() if group_filters is None else group_filters
+    def __call__(self, object_candidates):
+        # Keep any object whose states are satisfied
+        object_candidates[self._filter_name] = \
+            [obj for obj in object_candidates[self._filter_name] if self._op(obj.states[self._state].get_value(), self._val)]
 
-    def process(self, individual_objects, group_objects):
+        # Condition met if any object meets the condition
+        return len(object_candidates[self._filter_name]) > 0
+
+
+class ChangeCondition(RuleCondition):
+    """
+    Rule condition that checks all objects from @filter_name whether the sum of the prior conditions has changed
+    from the previous timestep
+    """
+    def __init__(
+            self,
+            filter_name,
+    ):
         """
-        Processes this transition rule at the current simulator step. If @condition evaluates to True, then
-        @transition will be executed.
+        Args:
+            filter_name (str): Name of the filter whose object candidates will be pruned based on whether or not any
+                prior conditions have changed or not
+        """
+        self._filter_name = filter_name
+        self._last_valid_candidates = None
+
+    def initialize(self, object_candidates):
+        # Initialize last valid candidates
+        self._last_valid_candidates = set(object_candidates[self._filter_name])
+
+    def __call__(self, object_candidates):
+        # Iterate over all current candidates -- if there's a mismatch in last valid candidates and current, then
+        # we store it, otherwise, we don't
+        objs = [obj for obj in object_candidates[self._filter_name] if obj not in self._last_valid_candidates]
+        object_candidates[self._filter_name] = objs
+        self._last_valid_candidates = set(objs)
 
+        # Valid if any object conditions have changed
+        return len(objs) > 0
```

Comment:
Sorry I am not sure I understand how this works. maybe we can chat in person?

Reply:
make change condition a wrapper